spec/bulk_processor_spec.rb

```diff
@@ -0,0 +1,140 @@
+include ActiveJob::TestHelper
+
+describe BulkProcessor do
+  describe '.process usage' do
+    let(:required_columns) { [] }
+    let(:optional_columns) { ['name'] }
+    let(:stream) { StringIO.new(csv) }
+    let(:csv) { "name\nRex" }
+
+    before do
+      TestItemProcessor.required_columns = required_columns
+      TestItemProcessor.optional_columns = optional_columns
```

Comment:
This sets global state on the class for the whole test suite. It'd be safer to create a subclass of `TestItemProcessor` right here with methods that return what you want.